File: qc-api/src/main/java/com/eager/questioncloud/api/creator/CreatorController.java
package com.eager.questioncloud.api.creator;

import com.eager.questioncloud.api.creator.Response.CreatorInformationResponse;
import com.eager.questioncloud.core.domain.creator.dto.CreatorDto.CreatorInformation;
import com.eager.questioncloud.core.domain.creator.service.CreatorService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.responses.ApiResponses;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/creator")
@RequiredArgsConstructor
public class CreatorController {
    private final CreatorService creatorService;

    @GetMapping("/{creatorId}")
    @ApiResponses(value = {
        @io.swagger.v3.oas.annotations.responses.ApiResponse(responseCode = "200", description = "요청 성공")
    })
    @Operation(operationId = "크리에이터 정보 조회", summary = "크리에이터 정보 조회", tags = {"creator"}, description = "크리에이터 정보 조회")
    public CreatorInformationResponse getCreatorInformation(@PathVariable Long creatorId) {
        CreatorInformation creatorInformation = creatorService.getCreatorInformation(creatorId);
        return new CreatorInformationResponse(creatorInformation);
    }
}

File: qc-core/qc-creator/qc-creator-api/src/main/kotlin/com/eager/questioncloud/creator/controller/CreatorController.kt
package com.eager.questioncloud.creator.controller

import com.eager.questioncloud.common.auth.UserPrincipal
import com.eager.questioncloud.creator.dto.CreatorInformationResponse
import com.eager.questioncloud.creator.dto.RegisterCreatorRequest
import com.eager.questioncloud.creator.dto.RegisterCreatorResponse
import com.eager.questioncloud.creator.service.CreatorInformationService
import com.eager.questioncloud.creator.service.RegisterCreatorService
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/creator")
class CreatorController(
    private val creatorInformationService: CreatorInformationService,
    private val registerCreatorService: RegisterCreatorService,
) {
    @PostMapping("/register")
    fun registerCreator(
        userPrincipal: UserPrincipal, @RequestBody request: @Valid RegisterCreatorRequest
    ): RegisterCreatorResponse {
        val creator = registerCreatorService.register(userPrincipal.userId, request.mainSubject, request.introduction)
        return RegisterCreatorResponse(creator.id)
    }
    
    @GetMapping("/{creatorId}")
    fun getCreatorInformation(@PathVariable creatorId: Long): CreatorInformationResponse {
        val creatorInformation = creatorInformationService.getCreatorInformation(creatorId)
        return CreatorInformationResponse(creatorInformation)
    }
}
